servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/appendPeriodToMembership.js

```diff
@@ -0,0 +1,66 @@
+const { Roles } = require('@orbiting/backend-modules-auth')
+const moment = require('moment')
+
+module.exports = async (_, { id, duration, durationUnit }, context) => {
+  const { user: me, t, req } = context
+  Roles.ensureUserHasRole(me, 'supporter')
+
+  const transaction = await context.pgdb.transactionBegin()
+
+  try {
+    const membership = await transaction.public.memberships.findOne({ id })
```

Comment:
Considering this mutation can run at any time, without knowledge of actions before, it should test some as `canManuallyAddPeriod`.

Reply:
How could I have forgotten that?!?